### ISSUE TITLE
App crashes when playing back any content

### ISSUE THREAD

**Issue by surrodox2001** [TeamNewPipe/NewPipe#4644]
### Version:
0.20.0 and 0.20.1

### Behavior:
1. open & play any content in any mode (fullscreen, popup or background) (doesn't matter its music or video)
2. program crashes instead of playing back content properly

NOTE: This issue doesn't happen when 'show notifications' is turned off in Android app setting

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Country:__ MY
* __Content Language:__ en
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.20.1
* __OS:__ Linux Android 5.1 - 22
<details><summary><b>Crash log </b></summary><p>

```
android.app.RemoteServiceException: Bad notification posted from package org.schabi.newpipe: Couldn't expand RemoteViews for: StatusBarNotification(pkg=org.schabi.newpipe user=UserHandle{0} id=123789 tag=null score=10 key=0|org.schabi.newpipe|123789|null|10111: Notification(pri=1 contentView=org.schabi.newpipe/0x109007e vibrate=null sound=null defaults=0x0 flags=0x62 color=0xff616161 category=transport actions=5 vis=PUBLIC))
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1644)
        at android.os.Handler.dispatchMessage(Handler.java:111)
        at android.os.Looper.loop(Looper.java:194)
        at android.app.ActivityThread.main(ActivityThread.java:5660)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:963)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:758)
        at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)

```
</details>
<hr>

**Comment by UserX404** [comment]
Maybe the same problem.
After starting a video or background player my phone often stucks in a phone-locked loop, where i'm not to entermy lock pin within time and i'm not able to kill NewPipe. The only two possibilities are to wait for the track has finished or to restart the phone.

Sadly 0.19.8 aren't able to encypt links anymore but until that was it was working without issues.
Sony Xperia XZ2c
Android 10

**Comment by Stypox** [comment]
Is this consistent, i.e. always reproducible? This might contain an explanation: https://stackoverflow.com/questions/25317659/how-to-fix-android-app-remoteserviceexception-bad-notification-posted-from-pac#26024747

**Comment by Dakkaron** [comment]
Not sure if it's the same problem but I keep getting this error when trying to open videos:

## Exception
* __User Action:__ requested stream
* __Request:__ https://www.youtube.com/watch?v=hp7JcmwKQcU
* __Content Country:__ GB
* __Content Language:__ en
* __App Language:__ de_AT
* __Service:__ YouTube
* __Version:__ 0.20.1
* __OS:__ Linux Android 7.1.1 - 25
<details><summary><b>Crash log </b></summary><p>

```
org.schabi.newpipe.extractor.exceptions.ParsingException: YouTube did not provide player config even after three attempts
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:664)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:56)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:68)
	at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(lambda)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14935)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
	at java.lang.Thread.run(Thread.java:761)

```
</details>
<hr>

**Comment by opusforlife2** [comment]
@Dakkaron Update.

**Comment by Dakkaron** [comment]
> @Dakkaron Update.

I've checked F-Droid daily since that bug first occured, but so far there is no new version...

**Comment by opusforlife2** [comment]
You'll have to wait. Or you can use our custom repo linked on the website.

**Comment by ShareASmile** [comment]
@Dakkaron You can also add NewPipe F-Droid repository to your fdroid client to receive updates even faster. Just add the following link to repositories in fdroid client. UPDATED Version will be shown.
https://archive.newpipe.net/fdroid/repo/
Take Help from
https://newpipe.schabi.org/FAQ/tutorials/install-add-fdroid-repo/

**Comment by inducer** [comment]
> https://archive.newpipe.net/fdroid/repo/

Thanks for the tip! Though that repo doesn't seem to be responding to HTTP requests for me. Any tips?

**Comment by opusforlife2** [comment]
@inducer The website server is down right now, including the repo.

**Comment by LeirbagGab** [comment]
I must say i seem to have the same issue on my S20+ running Android 10
`{"user_action":"requested stream","request":"https://www.youtube.com/watch?v=bmY3YQZQuPQ","content_language":"fr-FR","content_country":"FR","app_language":"fr_FR","service":"YouTube","package":"org.schabi.newpipe","version":"0.20.1","os":"Linux Android 10 - 29","time":"2020-11-09 13:41","exceptions":["org.schabi.newpipe.extractor.exceptions.ParsingException: YouTube did not provide player config even after three attempts\n\tat org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:664)\n\tat org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:56)\n\tat org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:68)\n\tat org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:64)\n\tat org.schabi.newpipe.util.ExtractorHelper.lambda$getStreamInfo$3(ExtractorHelper.java:116)\n\tat org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$5fJcha6Sq5APJBLdG6osaJby-mc.call(Unknown Source:4)\n\tat io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)\n\tat io.reactivex.Maybe.subscribe(Maybe.java:4290)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)\n\tat io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)\n\tat io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14935)\n\tat io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)\n\tat io.reactivex.Maybe.subscribe(Maybe.java:4290)\n\tat io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)\n\tat io.reactivex.Single.subscribe(Single.java:3666)\n\tat io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)\n\tat io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)\n\tat io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)\n\tat java.util.concurrent.FutureTask.run(FutureTask.java:266)\n\tat java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\n\tat java.lang.Thread.run(Thread.java:919)\n"],"user_comment":""}`

**Comment by opusforlife2** [comment]
@LeirbagGab Update.

**Comment by LeirbagGab** [comment]
@opusforlife2 any link to the latest apk? I only see my version on the website...

**Comment by opusforlife2** [comment]
Which website? Latest APK can be found on the Github Releases page, or from our custom F-Droid repo. Read https://github.com/TeamNewPipe/NewPipe#updates

**Comment by LeirbagGab** [comment]
Thx @opusforlife2

**Comment by Stypox** [comment]
Apparently this seems to be caused by a bug in the `NotificationCompat` implementation, as it can't originate from NewPipe's code [(according to SO this exception is thrown when some forbidden views are used inside the notification, but we have no control over those, only the library has)](https://stackoverflow.com/questions/1198260/how-can-you-iterate-over-the-elements-of-an-stdtuple/54225452#54225452) unless we messed up something really badly (and in that case it should happen on any phone). Maybe it was fixed by the dependency update PR, #4768, could you test it @surrodox2001 ? I'm linking it for now, as to not forget, it can always be de-linked
